src/pages/workspaces/workspace/analysis/modals/HailBatchModal.js

```diff
@@ -0,0 +1,151 @@
+import _ from 'lodash/fp';
+import { Fragment, useState } from 'react';
+import { div, h } from 'react-hyperscript-helpers';
+import { ButtonOutline, ButtonPrimary, Link, spinnerOverlay } from 'src/components/common';
+import { icon } from 'src/components/icons';
+import { withModalDrawer } from 'src/components/ModalDrawer';
+import TitleBar from 'src/components/TitleBar';
+import { Ajax } from 'src/libs/ajax';
```

Comment:
As opposed to using Ajax here, I'd recommend importing the individual modules where applicable.  Namely, `Apps`. It will make mocking calls out in tests a lot easier to not have an additional layer of misdirection